nomos-services/chain/chain-service/src/lib.rs

```diff
@@ -752,24 +773,40 @@ where
                                     &self.lib_subscription_sender,
                                     &self.service_resources_handle.state_updater,
-                                ).await {
-                            Ok((new_cryptarchia, new_storage_blocks_to_remove)) => {
-                                cryptarchia = new_cryptarchia;
-                                storage_blocks_to_remove = new_storage_blocks_to_remove;
-
-                                orphan_downloader.remove_orphan(&block.header().id());
-
-                                info!(counter.consensus_processed_blocks = 1);
-                            }
-                            Err(Error::Ledger(nomos_ledger::LedgerError::ParentNotFound(parent))
-                                | Error::Consensus(cryptarchia_engine::Error::ParentMissing(parent)),) => {
-
-                                orphan_downloader.enqueue_orphan(block.header().id(), cryptarchia.tip(), cryptarchia.lib());
-
-                                error!(target: LOG_TARGET, "Received block with parent {:?} that is not in the ledger state. Ignoring block.", parent);
+                                )
+                                .await
+                                {
+                                    Ok((new_cryptarchia, new_storage_blocks_to_remove)) => {
+                                        cryptarchia = new_cryptarchia;
+                                        storage_blocks_to_remove = new_storage_blocks_to_remove;
+
+                                        orphan_downloader.remove_orphan(&block.header().id());
+
+                                        info!(counter.consensus_processed_blocks = 1);
+                                    }
+                                    Err(
+                                        Error::Ledger(nomos_ledger::LedgerError::ParentNotFound(parent))
+                                            | Error::Consensus(cryptarchia_engine::Error::ParentMissing(parent)),
+                                    ) => {
+                                        orphan_downloader.enqueue_orphan(
+                                            block.header().id(),
+                                            cryptarchia.tip(),
+                                            cryptarchia.lib(),
+                                        );
+
+                                        error!(
+                                            target: LOG_TARGET,
+                                            "Received block with parent {:?} that is not in the ledger state. Ignoring block.",
+                                            parent
+                                        );
+                                    }
+                                    Err(e) => {
+                                        error!(target: LOG_TARGET, "Error processing reconstructed block: {:?}", e);
+                                    }
+                                }
```

Comment:
This is way too confusing to be honest. Its ok for now, but eventually it would be nice to have some meaningful functions for this.